modules/Rock.cmake

```diff
@@ -500,6 +534,18 @@ function(rock_prepare_pkgconfig TARGET_NAME DO_INSTALL)
     set(PKGCONFIG_CFLAGS ${${TARGET_NAME}_PKGCONFIG_CFLAGS})
     set(PKGCONFIG_LIBS ${${TARGET_NAME}_PKGCONFIG_LIBS})
 
+    if (DEFINED ROCK_PUBLIC_CXX_STANDARD)
+        set(cxx_standard ${ROCK_PUBLIC_CXX_STANDARD})
+    elseif (CMAKE_VERSION VERSION_LESS "3.1")
+        set(cxx_standard ${CMAKE_CXX_STANDARD})
+    else()
+        get_property(cxx_standard TARGET ${TARGET_NAME} PROPERTY CXX_STANDARD)
+    endif()
+
+    if (cxx_standard)
+        set(PKGCONFIG_CFLAGS "${PKGCONFIG_CFLAGS} -std=c++${cxx_standard}")
```

Comment:
Since CXX_EXTENSIONS will be on by default the hard setting of -std=c++11 and forwarding of this flag via pkg-config can conflict with the actual used compile flag -gnu++11, e.g. triggering issues like base-numeric#18 

(see also https://cmake.org/cmake/help/v3.1/manual/cmake-compile-features.7.html#manual:cmake-compile-features(7) for reference)

Obviously one option would be to disable CXX_EXTENSIONS by default. @doudou: comments?